lib/patch-by-id.js

```diff
@@ -0,0 +1,187 @@
+// Copyright IBM Corp. 2015,2016. All Rights Reserved.
+// Node module: loopback-datasource-juggler
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+var ValidationError = require('./validations').ValidationError;
+var assert = require('assert');
+var dao = require('./dao');
```

Comment:
I see we already have `lib/utils.js`, so we can either move the helpers there, but since they are DAO-specific, I think `lib/dao/helpers.js` would be better. Just keep the filename consistent with what we already have (plural `helpers.js`).